lib/Command/ConfigReport.php

```diff
@@ -63,12 +64,39 @@ public function run(InputInterface $input, OutputInterface $output) {
 	protected function configure() {
 		$this
 			->setName('configreport:generate')
-			->setDescription('generates a configreport');
+			->setDescription('generates a configreport')
+			->addOption('web', 'w', InputOption::VALUE_NONE, 'Get web config-report');
 	}
 
 	protected function execute(InputInterface $input, OutputInterface $output) {
+		if ($input->getOption('web')) {
+			$report = $this->getWebReport();
+			$output->writeln($report);
+			return 0;
+		}
+
 		$report = $this->reportDataCollector->getReportJson();
 		$output->writeln($report);
 		return 0;
 	}
+
+	private function getWebReport() {
+		$cfg = \OC::$server->getConfig();
+		$token = \bin2hex(\random_bytes(32));
+
+		$cfg->setAppValue('configreport', 'token', $token);
+
+		$urlGen = \OC::$server->getURLGenerator();
+		$url = $urlGen->linkToRouteAbsolute('configreport.Report.fromCli');
+
+		$opts = ['http' => [
```

Comment:
https?

Reply:
also wondering, wont we have here some problems as with federated shares and certificates ? @jvillafanez thoughs?

Reply:
e.g. that http is disallowed or https is allowed but only in certain ip ranges?

Reply:
For the http key, it seems both http and https share the same key for the context creation that will be used with the `file_get_contents` call (https://www.php.net/manual/en/context.http.php). Whether it actually uses https should depend only on the target url, so in terms of building the url, it should be ok.

For the certificates, it might be better if we skip the verification. At least it would give us less headaches. I'm not sure how though. Maybe adding `verify_peer => false` as part of the http key context, or in its own ssl key context.